kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/DataBrokerConnection.kt

```diff
@@ -52,6 +48,9 @@ class DataBrokerConnection internal constructor(
 ) {
     val disconnectListeners = MultiListener<DisconnectListener>()
 
+    private val dataBrokerApiInteraction = DataBrokerApiInteraction(managedChannel, dispatcher)
```

Comment:
How can we mock these two dependencies now in tests? For integration tests it works of course. Maybe we can move the dependencies to the constructor? Maybe introducing a "DataBrokerFetcher" could help removing the managed channel here. 


Regarding the managedChannel: Feels a bit error prone that the DataBrokerApiInteraction requires an opened managedChannel to actually work. Maybe we we need multiple channels and every ApiInteractor needs its own channel if possible?

Reply:
> How can we mock these two dependencies now in tests? For integration tests it works of course. Maybe we can move the dependencies to the constructor?

Agreed

> Feels a bit error prone that the DataBrokerApiInteraction requires an opened managedChannel to actually work.

I see your point but I'm ambivalent here. The class is used inside the DataBrokerConnection and you only get a DataBrokerConnection (in the normal flow) when the connection is already opened up by the connector. I can add some documentation to the class to make sure it's documented or add some additional checks / error handling. But I don't think it's as error prone as you say. 

> Maybe we we need multiple channels and every ApiInteractor needs its own channel if possible?
Let's not think about moving the connection logic inside there yet. It does not belong there. If multiple channels are needed, they can be provided by the constructor, just somebody else needs to open and provide them. We don't even know yet if you can't open multiple channels to the same "vehicle".

Reply:
Added Documentation and a check if the ManagedChannel is ready / connected.